### PR TITLE
SONAR-9156 use org’s default visibility in analysis, if prj not stored

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

**Comment by bartfastiel** [comment]
As also mentioned in the jira ticket, we are trying to solve this issue differently: The case of wanting to store components of a not yet stored project should never appear anyway, so we will just fail in that case.

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStep.java [hunk 1]
```diff
@@ -118,9 +119,16 @@ private void ensureConsistentVisibility(DbSession dbSession, String projectUuid,
...
 
-  private static boolean isRootPrivate(String rootKey, Map<String, ComponentDto> existingDtosByKeys) {
-    ComponentDto projectDto = existingDtosByKeys.get(rootKey);
-    return projectDto == null ? false /*FIXME actually use default configured for current organization*/ : projectDto.isPrivate();
+  private static boolean isRootPrivate(Component root, Map<String, ComponentDto> existingDtosByKeys) {
+    String rootKey = root.getKey();
+    ComponentDto rootDto = existingDtosByKeys.get(rootKey);
+    if (rootDto == null) {
+      if (Component.Type.VIEW == root.getType()) {
+        return NEW_VIEWS_SHOULD_BE_PRIVATE;
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStep.java]: Using `false` is enough here, without constant

File: server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStepTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    ComponentDao componentDao = mock(ComponentDao.class);
+    String projectKey = random(20);
+
+    doReturn(component).when(treeRootHolder).getRoot();
+    doReturn(projectKey).when(component).getKey();
+    doReturn(componentDao).when(dbClient).componentDao();
+    doReturn(emptyList()).when(componentDao).selectAllComponentsFromProjectKey(any(DbSession.class), eq(projectKey));
+
+    thrown.expect(IllegalStateException.class);
+    thrown.expectMessage("The project '"+projectKey+"' is not stored in the database, during a project analysis");
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStepTest.java]: missing whitespace :-)

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStep.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStep.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStepTest.java (hunk 2 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/PersistComponentsStepTest.java]: Done.

**Comment by bartfastiel** [comment]
This PR is ready to merge, as soon as QA is green.

**Pull request merged**